### DISCUSSION TITLE
[Chapter 03] 스프링 빈 스코프 중 프로토타입의 활용예시

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#17] [Chapter 03. Spring Application Basics] [answered]
여러 스프링 프로젝트를 진행하면서 기본 값인 singleton 빈만을 사용해 왔습니다.
prototype 빈을 사용해본 경험이 없는데, 사용해보신 경험이 있다면 어떠한 경우에 사용하는지, 예를 알려주시면 감사하겠습니다.

**Comment by ahngilwoong** [comment, marked as answer]
제가 프로젝트를 할때의 요구사항이었습니다.
창고의 크레인이 동시에 여러 쓰레드에서 실행되면서(1개의 창고당 3개의 크레인) 서로의 동작에 영향을 받지 않게 관리해야 했습니다.
View에서는 각각의 크레인마다의 CRUD요청이 존재했고
크레인의 행동에 대한 로깅, 트랜잭션등의 처리도 필요했습니다.
크레인의 개수도 한정적이며 사실 메모리 관점에서 크게 신경을 쓰지 않아서 가능했던 것 같습니다.
